src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeConstructorInfo.CoreCLR.cs

```diff
@@ -84,6 +94,34 @@ Signature LazyCreateSignature()
         private RuntimeType ReflectedTypeInternal => m_reflectedTypeCache.GetRuntimeType();
 
         internal BindingFlags BindingFlags => m_bindingFlags;
+
+
+        [DebuggerStepThrough]
+        [DebuggerHidden]
+        internal unsafe object InvokeNonEmitUnsafe(object? obj, IntPtr* args, Span<object?> argsForTemporaryMonoSupport, BindingFlags invokeAttr)
+        {
+            if ((invokeAttr & BindingFlags.DoNotWrapExceptions) == 0)
+            {
+                bool rethrow = false;
```

Comment:
Is the `rethrow` flag still needed? All interesting error handling should be done outside this method now so it should be safe to always wrap.

Reply:
Yes because of the call to `InvokeArrayConstructor()` which can throw due to invalid values, and we don't want to wrap that. There are a few more throws in ::InvokeMethod related to validation which we don't want to wrap because they don't come from the target user code via the invoked method, although with the managed code now doing a lot of the validation it is possible some of those are no longer reachable.

Reply:
> although with the managed code now doing a lot of the validation it is possible some of those are no longer reachable.

Right, all validation should be done in the managed code before we call the low-level unsafe invoke method.

In particular, validation of arguments for `InvokeArrayConstructor` should be taken care of by the managed `CheckArguments` method now.

Reply:
IIRC array ctor has an int-based "length" type which allows negative values so the native code checked that and threw. I'll double check that and if so see if we can verify that in managed code instead.

Reply:
The check for negative length is part of the array constructor implementation. We wrap similar argument validation checks for all other constructors and methods.

If we are not wrapping the exception from negative length arg validation for array constructors, it is an inconsistency/bug. I think it is fine to start wrapping that exception.

Reply:
Yes that makes sense - it's the method invoked that's the issue, not the callee's parameters. This happens for ctors when a negative int value causes an OverflowException.

The same type of issue occurs when a method returning a byref returns `null` which is not expected. This is thrown from the native `::InvokeMethod()`.

So I'll change these tests and add the changed semantics, along with the null+byref-like change, in a breaking change notification.